haystack/document_stores/pinecone.py

```diff
@@ -254,28 +324,34 @@ def write_documents(
         )
         if len(document_objects) > 0:
             add_vectors = False if document_objects[0].embedding is None else True
+            # If not adding vectors we use document namespace
+            namespace = self.embedding_namespace if add_vectors else self.document_namespace
+            if not add_vectors:
+                # To store documents in Pinecone, we use dummy embeddings (to be replaced with real embeddings later)
+                embeddings_to_index = np.zeros((batch_size, self.embedding_dim), dtype="float32")
+                # Convert embeddings to list objects
+                embeddings = [embed.tolist() if embed is not None else None for embed in embeddings_to_index]
```

Comment:
Yes, Pinecone cannot store documents without embeddings, so we just create a "document-only" namespace (essentially a separate document store/vector database) and add the documents with zero-vectors (eg they all look like [0.0, 0.0, 0.0,...])

At some point in the future there will be a feature for storing data without the vectors in Pinecone, at that point we'll probably switch to that rather than the two namespaces

Reply:
Thank you! I'm sure Pinecone is not the only docstore with this limitation. I'll keep track of this specific workaround as an example of where our DocStore API are not flexible enough :+1: